File: MyCallBagEyeNeurology/app/src/main/java/com/opticpro/suite/navigation/ToolsCatalog.kt
// Copyright (c) Roberto Di Flumeri
package com.opticpro.suite.navigation

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccessTime
import androidx.compose.material.icons.filled.Accessibility
import androidx.compose.material.icons.filled.Air
import androidx.compose.material.icons.filled.Architecture
import androidx.compose.material.icons.filled.Balance
import androidx.compose.material.icons.filled.Biotech
import androidx.compose.material.icons.filled.BlurCircular
import androidx.compose.material.icons.filled.Bloodtype
import androidx.compose.material.icons.filled.Calculate
import androidx.compose.material.icons.filled.CameraAlt
import androidx.compose.material.icons.filled.CenterFocusStrong
import androidx.compose.material.icons.filled.Circle
import androidx.compose.material.icons.filled.Contrast
import androidx.compose.material.icons.filled.Explore
import androidx.compose.material.icons.filled.Face
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.FlashOn
import androidx.compose.material.icons.filled.GridOn
import androidx.compose.material.icons.filled.Lightbulb
import androidx.compose.material.icons.filled.LocalHospital
import androidx.compose.material.icons.filled.MonitorHeart
import androidx.compose.material.icons.filled.MonitorWeight
import androidx.compose.material.icons.filled.Palette
import androidx.compose.material.icons.filled.PhotoCamera
import androidx.compose.material.icons.filled.Psychology
import androidx.compose.material.icons.filled.RemoveRedEye
import androidx.compose.material.icons.filled.Science
import androidx.compose.material.icons.filled.Straighten
import androidx.compose.material.icons.filled.Timeline
import androidx.compose.material.icons.filled.TrackChanges
import androidx.compose.material.icons.filled.ViewStream
import androidx.compose.material.icons.filled.ViewWeek
import androidx.compose.material.icons.filled.Visibility
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material.icons.filled.WaterDrop
import androidx.annotation.DrawableRes
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import com.opticpro.suite.R

data class ToolItem(val title: String, val route: String, val icon: ImageVector)

data class ToolCategory(
    val id: String,
    val title: String,
    val tools: List<ToolItem>,
    val icon: ImageVector,
    val accentColor: Color,
    @DrawableRes val headerImage: Int
)

/**
 * Catalogo completo degli strumenti, organizzato per reparto.
 * Ogni "route" punta a uno schermo reale registrato in AppNavGraph.
 */
object ToolsCatalog {

    val optometria = ToolCategory(
        id = "optometria",
        title = "Optometria",
        icon = Icons.Filled.RemoveRedEye,
        accentColor = Color(0xFF4FC3F7),
        headerImage = R.drawable.header_optometria,
        tools = listOf(
            ToolItem("Tabella acuità visiva (Vision Chart)", "vision_chart", Icons.Filled.Visibility),
            ToolItem("Misurazione distanza interpupillare (PD)", "pd_measurement", Icons.Filled.Straighten),
            ToolItem("Amsler Grid", "amsler_grid", Icons.Filled.GridOn),
            ToolItem("Duochrome Test", "duochrome", Icons.Filled.Contrast),
            ToolItem("Contrast Sensitivity Chart", "contrast_sensitivity", Icons.Filled.BlurCircular),
            ToolItem("Worth 4 Dot Test", "worth_4_dot", Icons.Filled.BlurCircular),
            ToolItem("Anisometropia Tool", "anisometropia", Icons.Filled.Balance),
            ToolItem("Crowding bars (ottotipo singolo)", "crowding_bars", Icons.Filled.ViewWeek),
            ToolItem("Near point of accommodation", "near_point_accommodation", Icons.Filled.CenterFocusStrong),
        )
    )

    val oftalmologia = ToolCategory(
        id = "oftalmologia",
        title = "Oftalmologia",
        icon = Icons.Filled.Visibility,
        accentColor = Color(0xFFBA68C8),
        headerImage = R.drawable.header_oftalmologia,
        tools = listOf(
            ToolItem("Lensometro AI (fotocamera)", "lensometer_ai", Icons.Filled.CameraAlt),
            ToolItem("Astigmatism Dial", "astigmatism_dial", Icons.Filled.Explore),
            ToolItem("Ishihara / Test colori", "color_vision_test", Icons.Filled.Palette),
            ToolItem("Topografia con disco di Placido", "placido_topography", Icons.Filled.TrackChanges),
            ToolItem("Esoftalmometro", "exophthalmometer", Icons.Filled.Straighten),
            ToolItem("Refrazione approssimata al comodino", "refraction_bedside", Icons.Filled.Biotech),
            ToolItem("Ullman indiretto (funduscopia)", "fundus_photography", Icons.Filled.PhotoCamera),
            ToolItem("Fotografia lampada a fessura", "slit_lamp_photography", Icons.Filled.PhotoCamera),
            ToolItem("Ricercatore asse per IOL toriche", "axis_finder_iol", Icons.Filled.Explore),
            ToolItem("Test strabismo selfie", "strabismus_selfie", Icons.Filled.Face),
        )
    )

    val neurologia = ToolCategory(
        id = "neurologia",
        title = "Neurologia",
        icon = Icons.Filled.Psychology,
        accentColor = Color(0xFFFFB74D),
        headerImage = R.drawable.header_neurologia,
        tools = listOf(
            ToolItem("Test Clock Drawing", "clock_drawing", Icons.Filled.AccessTime),
            ToolItem("NIH Stroke Scale", "nih_stroke_scale", Icons.Filled.MonitorHeart),
            ToolItem("Vestibular Ocular Motor Screening", "voms", Icons.Filled.RemoveRedEye),
            ToolItem("Penlight (torcia + documentazione foto)", "penlight", Icons.Filled.FlashOn),
            ToolItem("OKN Drum", "okn_drum", Icons.Filled.ViewStream),
            ToolItem("Mappa dei dermatomeri", "dermatome_map", Icons.Filled.Accessibility),
        )
    )

    val urgenza = ToolCategory(
        id = "urgenza",
        title = "Medicina d'urgenza",
        icon = Icons.Filled.LocalHospital,
        accentColor = Color(0xFFE57373),
        headerImage = R.drawable.header_urgenza,
        tools = listOf(
            ToolItem("Calibro pupillare", "pupil_gauge", Icons.Filled.Circle),
            ToolItem("Luce blu cobalto (abrasioni corneali)", "cobalt_blue_light", Icons.Filled.Lightbulb),
            ToolItem("Vein finder", "vein_finder", Icons.Filled.Bloodtype),
            ToolItem("Goniometro", "goniometro", Icons.Filled.Architecture),
        )
    )

    val calcolatori = ToolCategory(
        id = "calcolatori",
        title = "Calcolatori clinici",
        icon = Icons.Filled.Calculate,
        accentColor = Color(0xFF81C784),
        headerImage = R.drawable.header_calcolatori,
        tools = listOf(
            ToolItem("BMI Calculator", "calc_bmi", Icons.Filled.MonitorWeight),
            ToolItem("CHA2DS2-VASc Score", "calc_chads_vasc", Icons.Filled.Favorite),
            ToolItem("Glasgow Coma Scale (GCS)", "calc_gcs", Icons.Filled.Psychology),
            ToolItem("QTc (Bazett)", "calc_qtc", Icons.Filled.Timeline),
            ToolItem("Cockcroft-Gault (Clearance creatinina)", "calc_cockcroft_gault", Icons.Filled.WaterDrop),
            ToolItem("CKD-EPI (GFR)", "calc_ckd_epi", Icons.Filled.Science),
            ToolItem("Wells' Criteria for DVT", "calc_wells_dvt", Icons.Filled.Warning),
            ToolItem("Wells' Criteria for PE", "calc_wells_pe", Icons.Filled.Air),
        )
    )

    val all = listOf(optometria, oftalmologia, neurologia, urgenza, calcolatori)
}

File: MyCallBagEyeNeurologyIOS/OpticProSuite/Navigation/ToolsCatalog.swift
// © Roberto Di Flumeri
import SwiftUI

struct ToolItem: Identifiable, Hashable {
    let title: String
    let route: String
    let icon: String // SF Symbol name
    var id: String { route }
}

struct ToolCategory: Identifiable, Hashable {
    let id: String
    let title: String
    let tools: [ToolItem]
    let icon: String
    let accentColor: Color
    let headerImage: String
}

/// Catalogo completo degli strumenti, organizzato per reparto.
/// Ogni "route" punta a una schermata reale gestita in AppRootView.
enum ToolsCatalog {

    static let optometria = ToolCategory(
        id: "optometria",
        title: "Optometria",
        tools: [
            ToolItem(title: "Tabella acuità visiva (Vision Chart)", route: "vision_chart", icon: "eye"),
            ToolItem(title: "Misurazione distanza interpupillare (PD)", route: "pd_measurement", icon: "ruler"),
            ToolItem(title: "Amsler Grid", route: "amsler_grid", icon: "grid"),
            ToolItem(title: "Duochrome Test", route: "duochrome", icon: "circle.lefthalf.filled"),
            ToolItem(title: "Contrast Sensitivity Chart", route: "contrast_sensitivity", icon: "circle.dotted"),
            ToolItem(title: "Worth 4 Dot Test", route: "worth_4_dot", icon: "circle.grid.2x2"),
            ToolItem(title: "Anisometropia Tool", route: "anisometropia", icon: "scalemass"),
            ToolItem(title: "Crowding bars (ottotipo singolo)", route: "crowding_bars", icon: "rectangle.split.3x1"),
            ToolItem(title: "Near point of accommodation", route: "near_point_accommodation", icon: "scope"),
        ],
        icon: "eye",
        accentColor: Color(red: 0x4F / 255, green: 0xC3 / 255, blue: 0xF7 / 255),
        headerImage: "header_optometria"
    )

    static let oftalmologia = ToolCategory(
        id: "oftalmologia",
        title: "Oftalmologia",
        tools: [
            ToolItem(title: "Lensometro AI (fotocamera)", route: "lensometer_ai", icon: "camera"),
            ToolItem(title: "Astigmatism Dial", route: "astigmatism_dial", icon: "safari"),
            ToolItem(title: "Ishihara / Test colori", route: "color_vision_test", icon: "paintpalette"),
            ToolItem(title: "Topografia con disco di Placido", route: "placido_topography", icon: "target"),
            ToolItem(title: "Esoftalmometro", route: "exophthalmometer", icon: "ruler"),
            ToolItem(title: "Refrazione approssimata al comodino", route: "refraction_bedside", icon: "cross.case"),
            ToolItem(title: "Ullman indiretto (funduscopia)", route: "fundus_photography", icon: "camera.fill"),
            ToolItem(title: "Fotografia lampada a fessura", route: "slit_lamp_photography", icon: "camera.viewfinder"),
            ToolItem(title: "Ricercatore asse per IOL toriche", route: "axis_finder_iol", icon: "gyroscope"),
            ToolItem(title: "Test strabismo selfie", route: "strabismus_selfie", icon: "face.smiling"),
        ],
        icon: "eye",
        accentColor: Color(red: 0xBA / 255, green: 0x68 / 255, blue: 0xC8 / 255),
        headerImage: "header_oftalmologia"
    )

    static let neurologia = ToolCategory(
        id: "neurologia",
        title: "Neurologia",
        tools: [
            ToolItem(title: "Test Clock Drawing", route: "clock_drawing", icon: "clock"),
            ToolItem(title: "NIH Stroke Scale", route: "nih_stroke_scale", icon: "heart.text.square"),
            ToolItem(title: "Vestibular Ocular Motor Screening", route: "voms", icon: "eye"),
            ToolItem(title: "Penlight (torcia + documentazione foto)", route: "penlight", icon: "flashlight.on.fill"),
            ToolItem(title: "OKN Drum", route: "okn_drum", icon: "rectangle.grid.1x2"),
            ToolItem(title: "Mappa dei dermatomeri", route: "dermatome_map", icon: "figure.stand"),
        ],
        icon: "brain.head.profile",
        accentColor: Color(red: 0xFF / 255, green: 0xB7 / 255, blue: 0x4D / 255),
        headerImage: "header_neurologia"
    )

    static let urgenza = ToolCategory(
        id: "urgenza",
        title: "Medicina d'urgenza",
        tools: [
            ToolItem(title: "Calibro pupillare", route: "pupil_gauge", icon: "circle"),
            ToolItem(title: "Luce blu cobalto (abrasioni corneali)", route: "cobalt_blue_light", icon: "lightbulb"),
            ToolItem(title: "Vein finder", route: "vein_finder", icon: "drop"),
            ToolItem(title: "Goniometro", route: "goniometro", icon: "gauge"),
        ],
        icon: "cross.case.fill",
        accentColor: Color(red: 0xE5 / 255, green: 0x73 / 255, blue: 0x73 / 255),
        headerImage: "header_urgenza"
    )

    static let calcolatori = ToolCategory(
        id: "calcolatori",
        title: "Calcolatori clinici",
        tools: [
            ToolItem(title: "BMI Calculator", route: "calc_bmi", icon: "scalemass.fill"),
            ToolItem(title: "CHA2DS2-VASc Score", route: "calc_chads_vasc", icon: "heart.fill"),
            ToolItem(title: "Glasgow Coma Scale (GCS)", route: "calc_gcs", icon: "brain.head.profile"),
            ToolItem(title: "QTc (Bazett)", route: "calc_qtc", icon: "waveform.path.ecg"),
            ToolItem(title: "Cockcroft-Gault (Clearance creatinina)", route: "calc_cockcroft_gault", icon: "drop.fill"),
            ToolItem(title: "CKD-EPI (GFR)", route: "calc_ckd_epi", icon: "flask"),
            ToolItem(title: "Wells' Criteria for DVT", route: "calc_wells_dvt", icon: "exclamationmark.triangle"),
            ToolItem(title: "Wells' Criteria for PE", route: "calc_wells_pe", icon: "wind"),
        ],
        icon: "function",
        accentColor: Color(red: 0x81 / 255, green: 0xC7 / 255, blue: 0x84 / 255),
        headerImage: "header_calcolatori"
    )

    static let all: [ToolCategory] = [optometria, oftalmologia, neurologia, urgenza, calcolatori]
}
